### DISCUSSION TITLE
Compiling Stellarium with Kubuntu 22.10

### DISCUSSION THREAD

**Discussion by ilkant** [Stellarium/stellarium#2871] [Q&A]
I have compiled Stellarium 1.1 with Kubuntu 22.04 and it succeeded. And that Stellarium works in my computer. Then I have updated Kubuntu 22.04 -> 22.10. And now I get error message:

$ cmake -DCMAKE_INSTALL_PREFIX=/opt/stellarium ../.. 
-- Found CMake 3.24.2
CMake Error at /usr/lib/x86_64-linux-gnu/cmake/Qt6/QtFeature.cmake:1249 (message):
  Feature relocatable is already defined to be "ON" and should now be set to
  "OFF" when importing features from Qt6::Core.
Call Stack (most recent call first):
  /usr/lib/x86_64-linux-gnu/cmake/Qt6Core/Qt6CoreConfig.cmake:128 (qt_make_features_available)
  /usr/lib/x86_64-linux-gnu/cmake/Qt6/Qt6Config.cmake:223 (find_package)
  CMakeLists.txt:26 (FIND_PACKAGE)

I think that Kubuntu 22.10 have Qt 6 support. And it may be not good idea to edit and change /usr/lib/... files?

**Comment by 10110111** [comment]
Are you trying to build from an old build directory? If yes, try removing everything from the build directory (the one containing `CMakeCache.txt`) and running `cmake` again.

**Reply by ilkant** [reply]
I try to compile again in next week. I have other activities in weekend. I deleted all in build/unix directory and then:

$ cmake -DCMAKE_INSTALL_PREFIX=/opt/stellarium ../.. 
-- Found CMake 3.24.2
-- Platform: Linux-5.19.0-23-generic (x86_64)
-- Found Git 2.37.2
-- Building Stellarium 1.1 (v1.22.4-94821fc; Mode: Release)
-- Found Qt6: /usr/lib/qt6/bin/qmake (found suitable version 6.3.1)
-- Could NOT find Qt6Charts (missing: Qt6Charts_DIR)
CMake Error at CMakeLists.txt:580 (FIND_PACKAGE):
  Found package configuration file:

    /usr/lib/x86_64-linux-gnu/cmake/Qt6/Qt6Config.cmake

  but it set Qt6_FOUND to FALSE so package "Qt6" is considered to be NOT
  FOUND.  Reason given by package:

  Failed to find Qt component "Charts".

  Expected Config file at
  "/usr/lib/x86_64-linux-gnu/cmake/Qt6Charts/Qt6ChartsConfig.cmake" does NOT
  exist

**Comment by ilkant** [comment]
Now compilation works. It was a rtfm question. And when I changed libqt6charts6-dev to qt6-charts-dev the compiling succeeded. Now I have compiled on my computer:

Stellarium 1.1
Version 1.22.4-94821fc [master]
Based on Qt 6.3.1
